_posts/2023-04-05-deep-learning-scientific-computing-R-torch/deep-learning-scientific-computing-R-torch.Rmd

```diff
@@ -0,0 +1,66 @@
+
+In part three, we move beyond deep learning, and explore how `torch` can figure in general mathematical or scientific applications. Prominent topics are regression using matrix decompositions, the Discrete Fourier Transform, and the Wavelet Transform. Again, the goal is to understand how it works, and why it works like that (at least, to some degree). That's why, here just like in part one, we code algorithms from scratch, before introducing the speed-optimized `torch` equivalents.
+
+Now that you know about the book's content, you may be asking:
+
+### Who's it for?
+
+In short, Deep Learning and Scientific Computing with R `torch` -- being the only comprehensive text, as of this writing, on this topic -- addresses a wide audience. The hope is that there's something in it for everyone (well, most everyone).
+
+If you've never used `torch`, nor any other deep-learning framework, starting right from the beginning is the thing to do. No prior knowledge of deep learning is expected. The assumption is that you know some basic R, and are familiar with machine-learning terms such as supervised vs. unsupervised learning, training-validation-test set, et cetera. Having worked through part one, you'll find that parts two and three -- independently -- continue right from where you left off.
+
+If, on the other hand, you do have basic experience with `torch` and/or other automatic-differentiation frameworks, and are mostly interested in applied deep learning, you may be inclined to skim part one, and go to part two, checking out the applications that interest you most (or just browse, looking for inspiration). The domain-dependent examples were chosen to be rather generic and straightforward, so as to have the code generalize to a whole range of similar applications.
+
+Finally, if it was the "scientific computing" in the title that caught your attention, I certainly hope that part three has something for you! (As the book's author, I may say that writing this part was an extremely satisfying, incredibly engaging experience.[^1]) Part three really is where it makes sense to talk of "browsing" -- its topics hardly depend on each other, just look around for what appeals to you ...
+
+[^1]: though challenging, as well (*though* ... or *because*?)
+
+To wrap up, then:
+
+### What do I get?
+
+Content-wise, I think I can consider this question answered. If there were other books on `torch` with R, I'd probably stress two things: First, the already-referred-to focus on concepts and understanding. Second, the usefulness of the code examples. By using off-the-shelf datasets, and performing the usual types of tasks, we write code fit to serve as a start in your own applications -- providing templates, sort of, ready to copy-paste and adapt to a purpose.
```

Comment:
```suggestion
Content-wise, I think I can consider this question answered. If there were other books on `torch` with R, I'd probably stress two things: First, the already-referred-to focus on concepts and understanding. Second, the usefulness of the code examples. By using off-the-shelf datasets, and performing the usual types of tasks, we write code fit to serve as a start in your own applications -- providing templates ready to copy-paste and adapt to a purpose.
```

Reply:
you're right, as always ... one should be :-)))